{
  "uid" : "c5340af7cf1a5e5b",
  "name" : "test_login_failed",
  "fullName" : "test_login#test_login_failed",
  "historyId" : "66652913f36bc6fd88d92a6bd7aca0c6",
  "time" : {
    "start" : 1714581481359,
    "stop" : 1714581485124,
    "duration" : 3765
  },
  "status" : "failed",
  "statusMessage" : "selene.core.exceptions.TimeoutException: Message: \n\nTimed out after 3s, while waiting for:\nbrowser.element(('xpath', '//*[@id=\"login_button_container\"]/div/form/h3')).is visible\n\nReason: NoSuchElementException: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//*[@id=\"login_button_container\"]/div/form/h3\"}\n  (Session info: chrome=124.0.6367.60); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nScreenshot: file:///home/runner/.selene/screenshots/1714581480152/1714581480152.png\nPageSource: file:///home/runner/.selene/screenshots/1714581480152/1714581480152.html",
  "statusTrace" : "fn = <selene.core.conditions.ElementCondition object at 0x7f5afa781290>\n\n    def logic(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n>               return fn(self.entity)\n\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/wait.py:119: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/condition.py:174: in __call__\n    return self._fn(entity)\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/condition.py:121: in fn\n    if not predicate(entity):\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/match.py:38: in <lambda>\n    'is visible', lambda element: element().is_displayed()\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/entity.py:192: in __call__\n    return self.locate()\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/entity.py:185: in locate\n    return self._locator()\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/locator.py:34: in __call__\n    return self._locate()\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/_browser.py:71: in <lambda>\n    Locator(f'{self}.element({by})', lambda: self.driver.find_element(*by)),\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:741: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7f5afa689b10>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"no such element: Unable to locate element: {\\...\\\\n#16 0x563e3af13600 \\\\u003Cunknown>\\\\n#17 0x563e3af22f84 \\\\u003Cunknown>\\\\n#18 0x7fa10fe94ac3 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//*[@id=\"login_button_container\"]/div/form/h3\"}\nE         (Session info: chrome=124.0.6367.60); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       #0 0x563e3af23e33 <unknown>\nE       #1 0x563e3ac13567 <unknown>\nE       #2 0x563e3ac5e27d <unknown>\nE       #3 0x563e3ac5e331 <unknown>\nE       #4 0x563e3aca1624 <unknown>\nE       #5 0x563e3ac8038d <unknown>\nE       #6 0x563e3ac9eae2 <unknown>\nE       #7 0x563e3ac80103 <unknown>\nE       #8 0x563e3ac511ba <unknown>\nE       #9 0x563e3ac51b8e <unknown>\nE       #10 0x563e3aee859b <unknown>\nE       #11 0x563e3aeec34b <unknown>\nE       #12 0x563e3aed4681 <unknown>\nE       #13 0x563e3aeeceb2 <unknown>\nE       #14 0x563e3aeb96af <unknown>\nE       #15 0x563e3af13428 <unknown>\nE       #16 0x563e3af13600 <unknown>\nE       #17 0x563e3af22f84 <unknown>\nE       #18 0x7fa10fe94ac3 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py:229: NoSuchElementException\n\nDuring handling of the above exception, another exception occurred:\n\n    def test_login_failed():\n        browser.open('https://www.saucedemo.com/v1/')\n        s('//*[@id=\"user-name\"]').type('standard_user')\n        s('//*[@id=\"password\"]').type('secret_sauce')\n        s('//*[@id=\"login-button\"]').click()\n>       s('//*[@id=\"login_button_container\"]/div/form/h3').should(be.visible)\n\ntest_login.py:17: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/entity.py:132: in should\n    self.wait.for_(condition)\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/wait.py:150: in for_\n    return self._decorator(self)(logic)(fn)\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/support/_logging.py:104: in decorated\n    return for_(fn)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nfn = <selene.core.conditions.ElementCondition object at 0x7f5afa781290>\n\n    def logic(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n                return fn(self.entity)\n            except Exception as reason:\n                if time.time() > finish_time:\n                    reason_string = '{name}: {message}'.format(\n                        name=reason.__class__.__name__,\n                        message=getattr(reason, \"msg\", str(reason)),\n                    )\n                    # TODO: think on how can we improve logging failures in selene, e.g. reverse msg and stacktrace\n                    # stacktrace = getattr(reason, 'stacktrace', None)\n                    # TODO: should we have an option to turn on stacktrace logging?\n                    timeout = self._timeout\n                    entity = self.entity\n    \n                    # if it's a normal function, it should have __qualname__,\n                    # then use it, otherwise use str(fn)\n                    fn_name = getattr(\n                        fn,\n                        '__qualname__',\n                        getattr(fn, '__str__', lambda: str(fn))(),\n                    )\n    \n                    failure = TimeoutException(\n                        f'\\n'\n                        f'\\nTimed out after {timeout}s, while waiting for:'\n                        f'\\n{entity}.{fn_name}'\n                        f'\\n'\n                        f'\\nReason: {reason_string}'\n                    )\n    \n>                   raise self._hook_failure(failure)\nE                   selene.core.exceptions.TimeoutException: Message: \nE                   \nE                   Timed out after 3s, while waiting for:\nE                   browser.element(('xpath', '//*[@id=\"login_button_container\"]/div/form/h3')).is visible\nE                   \nE                   Reason: NoSuchElementException: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//*[@id=\"login_button_container\"]/div/form/h3\"}\nE                     (Session info: chrome=124.0.6367.60); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE                   Screenshot: file:///home/runner/.selene/screenshots/1714581480152/1714581480152.png\nE                   PageSource: file:///home/runner/.selene/screenshots/1714581480152/1714581480152.html\n\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/wait.py:148: TimeoutException",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "browser_management",
    "time" : {
      "start" : 1714581481358,
      "stop" : 1714581481358,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "status" : "failed",
    "statusMessage" : "selene.core.exceptions.TimeoutException: Message: \n\nTimed out after 3s, while waiting for:\nbrowser.element(('xpath', '//*[@id=\"login_button_container\"]/div/form/h3')).is visible\n\nReason: NoSuchElementException: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//*[@id=\"login_button_container\"]/div/form/h3\"}\n  (Session info: chrome=124.0.6367.60); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nScreenshot: file:///home/runner/.selene/screenshots/1714581480152/1714581480152.png\nPageSource: file:///home/runner/.selene/screenshots/1714581480152/1714581480152.html",
    "statusTrace" : "fn = <selene.core.conditions.ElementCondition object at 0x7f5afa781290>\n\n    def logic(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n>               return fn(self.entity)\n\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/wait.py:119: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/condition.py:174: in __call__\n    return self._fn(entity)\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/condition.py:121: in fn\n    if not predicate(entity):\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/match.py:38: in <lambda>\n    'is visible', lambda element: element().is_displayed()\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/entity.py:192: in __call__\n    return self.locate()\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/entity.py:185: in locate\n    return self._locator()\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/locator.py:34: in __call__\n    return self._locate()\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/_browser.py:71: in <lambda>\n    Locator(f'{self}.element({by})', lambda: self.driver.find_element(*by)),\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:741: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7f5afa689b10>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"no such element: Unable to locate element: {\\...\\\\n#16 0x563e3af13600 \\\\u003Cunknown>\\\\n#17 0x563e3af22f84 \\\\u003Cunknown>\\\\n#18 0x7fa10fe94ac3 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//*[@id=\"login_button_container\"]/div/form/h3\"}\nE         (Session info: chrome=124.0.6367.60); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       #0 0x563e3af23e33 <unknown>\nE       #1 0x563e3ac13567 <unknown>\nE       #2 0x563e3ac5e27d <unknown>\nE       #3 0x563e3ac5e331 <unknown>\nE       #4 0x563e3aca1624 <unknown>\nE       #5 0x563e3ac8038d <unknown>\nE       #6 0x563e3ac9eae2 <unknown>\nE       #7 0x563e3ac80103 <unknown>\nE       #8 0x563e3ac511ba <unknown>\nE       #9 0x563e3ac51b8e <unknown>\nE       #10 0x563e3aee859b <unknown>\nE       #11 0x563e3aeec34b <unknown>\nE       #12 0x563e3aed4681 <unknown>\nE       #13 0x563e3aeeceb2 <unknown>\nE       #14 0x563e3aeb96af <unknown>\nE       #15 0x563e3af13428 <unknown>\nE       #16 0x563e3af13600 <unknown>\nE       #17 0x563e3af22f84 <unknown>\nE       #18 0x7fa10fe94ac3 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py:229: NoSuchElementException\n\nDuring handling of the above exception, another exception occurred:\n\n    def test_login_failed():\n        browser.open('https://www.saucedemo.com/v1/')\n        s('//*[@id=\"user-name\"]').type('standard_user')\n        s('//*[@id=\"password\"]').type('secret_sauce')\n        s('//*[@id=\"login-button\"]').click()\n>       s('//*[@id=\"login_button_container\"]/div/form/h3').should(be.visible)\n\ntest_login.py:17: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/entity.py:132: in should\n    self.wait.for_(condition)\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/wait.py:150: in for_\n    return self._decorator(self)(logic)(fn)\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/support/_logging.py:104: in decorated\n    return for_(fn)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nfn = <selene.core.conditions.ElementCondition object at 0x7f5afa781290>\n\n    def logic(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n                return fn(self.entity)\n            except Exception as reason:\n                if time.time() > finish_time:\n                    reason_string = '{name}: {message}'.format(\n                        name=reason.__class__.__name__,\n                        message=getattr(reason, \"msg\", str(reason)),\n                    )\n                    # TODO: think on how can we improve logging failures in selene, e.g. reverse msg and stacktrace\n                    # stacktrace = getattr(reason, 'stacktrace', None)\n                    # TODO: should we have an option to turn on stacktrace logging?\n                    timeout = self._timeout\n                    entity = self.entity\n    \n                    # if it's a normal function, it should have __qualname__,\n                    # then use it, otherwise use str(fn)\n                    fn_name = getattr(\n                        fn,\n                        '__qualname__',\n                        getattr(fn, '__str__', lambda: str(fn))(),\n                    )\n    \n                    failure = TimeoutException(\n                        f'\\n'\n                        f'\\nTimed out after {timeout}s, while waiting for:'\n                        f'\\n{entity}.{fn_name}'\n                        f'\\n'\n                        f'\\nReason: {reason_string}'\n                    )\n    \n>                   raise self._hook_failure(failure)\nE                   selene.core.exceptions.TimeoutException: Message: \nE                   \nE                   Timed out after 3s, while waiting for:\nE                   browser.element(('xpath', '//*[@id=\"login_button_container\"]/div/form/h3')).is visible\nE                   \nE                   Reason: NoSuchElementException: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//*[@id=\"login_button_container\"]/div/form/h3\"}\nE                     (Session info: chrome=124.0.6367.60); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE                   Screenshot: file:///home/runner/.selene/screenshots/1714581480152/1714581480152.png\nE                   PageSource: file:///home/runner/.selene/screenshots/1714581480152/1714581480152.html\n\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/wait.py:148: TimeoutException",
    "steps" : [ {
      "name" : "element('xpath', '//*[@id=\"user-name\"]'): type: standard_user",
      "time" : {
        "start" : 1714581481713,
        "stop" : 1714581481775,
        "duration" : 62
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "element('xpath', '//*[@id=\"password\"]'): type: secret_sauce",
      "time" : {
        "start" : 1714581481775,
        "stop" : 1714581481820,
        "duration" : 45
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "element('xpath', '//*[@id=\"login-button\"]'): click",
      "time" : {
        "start" : 1714581481820,
        "stop" : 1714581481952,
        "duration" : 132
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "element('xpath', '//*[@id=\"login_button_container\"]/div/form/h3'): should be visible",
      "time" : {
        "start" : 1714581481952,
        "stop" : 1714581485122,
        "duration" : 3170
      },
      "status" : "failed",
      "statusMessage" : "selene.core.exceptions.TimeoutException: Message: \n\nTimed out after 3s, while waiting for:\nbrowser.element(('xpath', '//*[@id=\"login_button_container\"]/div/form/h3')).is visible\n\nReason: NoSuchElementException: no such element: Unable to locate element: {\"method\":\"xpath\",\"selector\":\"//*[@id=\"login_button_container\"]/div/form/h3\"}\n  (Session info: chrome=124.0.6367.60); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nScreenshot: file:///home/runner/.selene/screenshots/1714581480152/1714581480152.png\nPageSource: file:///home/runner/.selene/screenshots/1714581480152/1714581480152.html\n\n",
      "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/support/_logging.py\", line 104, in decorated\n    return for_(fn)\n           ^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selene/core/wait.py\", line 148, in logic\n    raise self._hook_failure(failure)\n",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : true,
      "hasContent" : true
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 4,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ {
    "name" : "browser_management::0",
    "time" : {
      "start" : 1714581485204,
      "stop" : 1714581485420,
      "duration" : 216
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ {
      "uid" : "a092fcc0093bc9cf",
      "name" : "screenshot",
      "source" : "a092fcc0093bc9cf.png",
      "type" : "image/png",
      "size" : 383737
    } ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 1,
    "shouldDisplayMessage" : false,
    "hasContent" : true
  } ],
  "labels" : [ {
    "name" : "suite",
    "value" : "test_login"
  }, {
    "name" : "host",
    "value" : "fv-az1393-442"
  }, {
    "name" : "thread",
    "value" : "2323-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "test_login"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Product defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "tags" : [ ]
  },
  "source" : "c5340af7cf1a5e5b.json",
  "parameterValues" : [ ]
}